sqlglot/dialects/mysql.py

```diff
@@ -5,9 +5,61 @@
     no_paren_current_date_sql,
     no_tablesample_sql,
     no_trycast_sql,
+    date_add_interval,
 )
 from sqlglot.generator import Generator
+from sqlglot.helper import list_get, invert
 from sqlglot.parser import Parser
+from sqlglot.time import format_time
+
+
+def _date_trunc_sql(self, expression):
+    unit = expression.args.get("unit").name
+
+    this = self.sql(expression.this)
+
+    if unit.lower() == "day":
+        return f"DATE({this})"
+
+    if unit.lower() == "week":
+        concat = f"CONCAT(YEAR({this}), ' ', WEEK({this}, 1), ' 1')"
+        date_format = "%Y %u %w"
+    elif unit.lower() == "month":
+        concat = f"CONCAT(YEAR({this}), ' ', MONTH({this}), ' 1')"
+        date_format = "%Y %c %e"
+    elif unit.lower() == "quarter":
+        concat = f"CONCAT(YEAR({this}), ' ', QUARTER({this}) * 3 - 2, ' 1')"
+        date_format = "%Y %c %e"
+    elif unit.lower() == "year":
+        concat = f"CONCAT(YEAR({this}), ' 1 1')"
+        date_format = "%Y %c %e"
+    else:
+        self.unsupported("Unexpected interval unit: {unit}")
+        return f"DATE({this})"
+
+    return f"STR_TO_DATE({concat}, '{date_format}')"
+
+
+def _str_to_date(args):
+    date_format = format_time(list_get(args, 1).name, MySQL.time_mapping)
+    return exp.StrToDate(this=list_get(args, 0), format=exp.Literal.string(date_format))
+
+
+def _str_to_date_sql(self, expression):
+    date_format = format_time(
+        expression.args.get("format").name, invert(MySQL.time_mapping)
```

Comment:
MySQL.inverse_time_mapping already exists

Reply:
you can also pass in the time_trie 

should you be calling self.format_time though? format_time from time is the lower level method

Reply:
Ah, ok, I hadn't dug deep enough into format_time.

I think all the FUNCTION callables need a `self` parameter. Yikes.